spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/query/SqlSpannerQueryTests.java

```diff
@@ -496,18 +504,97 @@ public void sqlCountWithWhereTest() throws NoSuchMethodException {
 			assertThat(paramMap.get("id").getString()).isEqualTo(params[0]);
 			assertThat(paramMap.get("trader_id").getString()).isEqualTo(params[1]);
 
-			return null;
-		}).when(this.spannerTemplate).executeQuery(any(), any());
+			return Arrays.asList(2L);
+		}).when(this.spannerTemplate).query((Function<Struct, Object>) any(), any(), any());
```

Comment:
So this change is not because of the added json logic. I found when running this test, it is originally going through the wrong logic: in `executeReadSql`, because `sqlSpannerQuery.getReturnedSimpleConvertableItemType()` was not mocked to return `true`, it then takes the path to the query method returning the whole entity, which isn't the case in real sql count queries.  At the end of both routes, `this.spannerTemplate.executeQuery` get called with same parameters, so result as far as this test concerns is the same. But I wanted to make this change to reflect the actual code path that this query should take. 
That said, the changes here to verify on `query` instead of `executeQuery` is one layer up and not ideal. I made the compromise because the test breaks [here](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/206a2531a900ed4643483fc9341746a99ee43654/spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/SpannerTemplate.java#L239-L243) without mocking `resultSet` correctly. 
In a newer commit, I am able to switch back to verify `executeQuery` with `resultSet` mocked.